File: 2887-sort-vowels-in-a-string/sort-vowels-in-a-string.go
func sortVowels(s string) string {
	vowels := map[rune]bool{
		'a': true, 'e': true, 'i': true, 'o': true, 'u': true,
		'A': true, 'E': true, 'I': true, 'O': true, 'U': true,
	}

	// Step 1: Extract vowels
	var vowelList []rune
	for _, ch := range s {
		if vowels[ch] {
			vowelList = append(vowelList, ch)
		}
	}

	// Step 2: Sort vowels by ASCII
	sort.Slice(vowelList, func(i, j int) bool { return vowelList[i] < vowelList[j] })

	// Step 3: Replace vowels with sorted ones
	res := []rune(s)
	idx := 0
	for i, ch := range res {
		if vowels[ch] {
			res[i] = vowelList[idx]
			idx++
		}
	}

	// Step 4: Return result
	return string(res)
}

File: 2887-sort-vowels-in-a-string/sort-vowels-in-a-string.java
class Solution {
    public String sortVowels(String s) {
        Set<Character> vowels = new HashSet<>(Arrays.asList(
                'a', 'e', 'i', 'o', 'u', 'A', 'E', 'I', 'O', 'U'));

        // Step 1: Extract vowels
        List<Character> vowelList = new ArrayList<>();
        char[] arr = s.toCharArray();
        for (char ch : arr)
            if (vowels.contains(ch))
                vowelList.add(ch);

        // Step 2: Sort vowels by ASCII
        Collections.sort(vowelList);

        // Step 3: Replace vowels with sorted ones
        int idx = 0;
        for (int i = 0; i < arr.length; i++)
            if (vowels.contains(arr[i]))
                arr[i] = vowelList.get(idx++);

        // Step 4: Return result
        return new String(arr);
    }
}